Description: Make sure that all allocated buffers are freed regardless of driver
Author: Michael Sweet <msweet@apple.com>
Bug: https://www.cups.org/str.php?L4393
--- a/filter/rastertolabel.c
+++ b/filter/rastertolabel.c
@@ -639,13 +639,6 @@
 
         puts("^IDR:CUPS.GRF^FS");
 	puts("^XZ");
-
-       /*
-        * Free compression buffers...
-	*/
-
-	free(CompBuffer);
-	free(LastBuffer);
         break;
 
     case ZEBRA_CPCL :
@@ -714,6 +707,18 @@
   */
 
   free(Buffer);
+
+  if (CompBuffer)
+  {
+    free(CompBuffer);
+    CompBuffer = NULL;
+  }
+
+  if (LastBuffer)
+  {
+    free(LastBuffer);
+    LastBuffer = NULL;
+  }
 }
 
 
